pika/adapters/blocking_connection.py

```diff
@@ -1542,21 +1543,24 @@ def basic_consume(self,
             consumer_tag is already present.
 
         """
+        if 'no_ack' in kwargs:
+            auto_ack = bool(kwargs.get('no_ack'))
```

Comment:
Good catch, I wouldn't have thought to even look for that.

Reply:
Should `no_ack` be a kwarg in the function signatures then? I'm not sure how kwarg aliasing is typically handled in the Python community, sorry.

Reply:
@michaelklishin, I think that the common approach is to pop the expected args from `kwargs` and then raise an exception if kwargs is not empty. I would also get rid of the `bool` typecast since we don't apply it to `auto_ack` and we didn't have it for `no_ack` either. For example:
```
if 'no_ack' in kwargs:
    auto_ack = kwargs.pop('no_ack')

if kwargs: # pythonic way to check for empty container
    raise TypeError(
        'basic_consume called with {} unexpected argument(s): {}'.format(
            len(kwargs),
            compat.dictkeys(kwargs)))
```